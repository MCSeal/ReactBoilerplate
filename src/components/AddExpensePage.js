import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const AddExpensePage = () => (
    <div>
       add expense page!
    </div>
    
);

export default AddExpensePage;