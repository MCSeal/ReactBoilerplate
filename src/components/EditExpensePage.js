import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const EditExpensePage = () => (
    <div>
       edit
    </div>
    
);


export default EditExpensePage;