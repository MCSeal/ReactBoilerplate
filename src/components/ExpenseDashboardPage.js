import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';


const ExpenseDashboardPage = () => (
    <div>
        from the dashboard sl
    </div>
    
);

export default ExpenseDashboardPage;