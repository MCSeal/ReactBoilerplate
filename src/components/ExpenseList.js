import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';


export const ExpenseList = (props) => (
    <div>
        {
            
            props.expenses.length === 0 ? (
            <p>No expenses...yet</p>
            
            ) : (
            
            props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense}/>
            })
            
            )
            
        }
        
    </div>
);

const mapStateToProps = (state) => {
    return {
        
        //state.expenses would work too, this is for the filter
      expenses: selectExpenses(state.expenses, state.filters)
    };
}

//connect attaches redux and react
export default connect(mapStateToProps)(ExpenseList);

