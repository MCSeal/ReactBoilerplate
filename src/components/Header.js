
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';



const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}> Go home</NavLink>
        <NavLink to="/create" activeClassName="is-active"> Create expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active"> Edit page</NavLink>
        <NavLink to="/help" activeClassName="is-active"> Do you need help?</NavLink>
    </header>
    
);

export default Header;