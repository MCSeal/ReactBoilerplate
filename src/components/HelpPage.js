import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const HelpPage = () => (
    <div>
       help
    </div>
    
);

export default HelpPage;