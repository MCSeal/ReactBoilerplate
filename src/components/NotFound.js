import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

//similar to Href link is for react
const NotFound = () => (
    <div>
    404 not found!
       <Link to="/"> Go home</Link>
    </div>
    
);

export default NotFound;