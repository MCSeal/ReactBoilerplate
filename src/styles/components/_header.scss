.header {
    margin-bottom: $xl-size;
    text-align: center;


}

.header__title {
    margin-top: 30rem;
    font-size: $xl-size;
    font-family: 'Tangerine', arial;
}

.header__subtitle {
    font-size: $m-size;
    
}

