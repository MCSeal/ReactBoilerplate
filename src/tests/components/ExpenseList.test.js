import React from 'react';
import { ExpenseList } from '../../components/ExpenseList';
import { shallow } from 'enzyme';
import moment from 'moment';

const expenses = [{
    id: '1',
    description: 'Gum',
    note: '',
    amount: 195,
    createdAt: 0
},{
    id: '2',
    description: 'rent',
    note: '',
    amount: 109500,
    createdAt: moment(0).subtract(4, 'days').valueOf()
},{
    id: '1',
    description: 'credit card',
    note: '',
    amount: 4500,
    createdAt: moment(0).add(4, 'days').valueOf()
}]




test('should render expenlist', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
    
});


test('should render with empty array', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
    
});

