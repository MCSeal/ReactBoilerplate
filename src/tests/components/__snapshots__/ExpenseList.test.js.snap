// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenlist 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="credit card"
    id="1"
    key="1"
    note=""
  />
</div>
`;

exports[`should render with empty array 1`] = `
<div>
  <p>
    No expenses...yet
  </p>
</div>
`;
